feat: finance sec search environment (#1473)
NVIDIA-NeMo/Gym: environments/finance_sec_search/README.md

> # Finance SEC Search Resource Server

Reviewer: I think it would be good to point users to example files that we have in the `data` dir

**Only companies listed in the [SEC company tickers file](https://www.sec.gov/files/company_tickers.json) are supported.** Questions about companies not in this list will fail at the ticker lookup step.

## Tools

| Tool | Description |
|------|-------------|
| `sec_filing_search` | Search SEC EDGAR for filing metadata by stock ticker symbol |
| `parse_html_page` | Fetch and parse any HTML page (SEC URLs use disk cache), store under a key |
| `retrieve_information` | Query stored documents via LLM prompt with `{{key}}` placeholders |
| `submit_final_result` | Submit the final answer (keeps model in tool-calling mode until ready) |
| `web_search` | Internet search via Tavily API (optional — requires `tavily_api_key`) |

If `tavily_api_key` is not configured, `web_search` returns an error directing the model to use SEC tools instead.

## Setup

### env.yaml

Create `env.yaml` in the Gym root:

```yaml
policy_base_url: https://api.openai.com/v1
policy_api_key: empty
policy_model_name: gpt-5-mini

search_judge_model_base_url: https://api.openai.com/v1
search_judge_model_api_key: empty
search_judge_model_name: gpt-5-mini

> # Optional: enable web_search tool (requires Tavily API key)

Reviewer: ```suggestion
# Optional: enable web_search tool (requires Tavily API key). Set to `null` to run without web_search tool
```

Reviewer: I interpreted "optional" == can be skipped in the config, but this field is referenced so it needs to be set.

```

Convert raw questions into Gym input format (adds tool definitions, system prompt, etc.):

```bash
python environments/finance_sec_search/scripts/convert_questions.py \
  --input environments/finance_sec_search/data/example_questions.jsonl \
  --output environments/finance_sec_search/data/example.jsonl
```

Add `--include-web-search` / `-w` to include the optional `web_search` tool:

```bash
python environments/finance_sec_search/scripts/convert_questions.py \
  --input environments/finance_sec_search/data/example_questions.jsonl \
  --output environments/finance_sec_search/data/example.jsonl \
  --include-web-search
```

A pre-converted `example.jsonl` (without web search) is checked in and ready to
use — you only need to re-run `convert_questions.py` if you modify the raw
questions or want to change the tool set.

#### Vals AI public benchmark (`prepare.py`)

The [Vals AI finance-agent](https://github.com/vals-ai/finance-agent) 50-question
public benchmark lives in `benchmarks/finance_sec_search/`. It downloads the
`public.csv` dataset from GitHub and converts it to Gym format:

```bash
# Prepare via Gym CLI (recommended — used by ng_run with benchmark configs):

> ng_prepare_benchmark +benchmark=benchmarks/finance_sec_search/config_no_web_search.yaml

Reviewer: ```suggestion
ng_prepare_benchmark +config_paths=[benchmarks/finance_sec_search/config_no_web_search.yaml]
```
The syntax with `+benchmark` did not work for me and I don't know how to fix it

Reviewer: I also think it would be good to explicitly add a 2nd command with `config_web_search.yaml` here

# Or run the script directly:
python benchmarks/finance_sec_search/prepare.py            # without web_search
python benchmarks/finance_sec_search/prepare.py --include-web-search  # with web_search
```

Output is written to `benchmarks/finance_sec_search/data/`:

| Config | Prepare script | Output file |
|--------|---------------|-------------|
| `config_no_web_search.yaml` | `prepare.py` | `finance_sec_search_benchmark.jsonl` |
| `config_web_search.yaml` | `prepare_web_search.py` | `finance_sec_search_benchmark_web_search.jsonl` |

> **Note:** `prepare.py` duplicates the prompt and tool definitions from
> `convert_questions.py`. They are functionally identical — `convert_questions.py`
> is the canonical source for custom questions, while `prepare.py` is specific to
> downloading and converting the Vals AI dataset.

#### SecQue benchmark

To prepare the [SecQue](https://huggingface.co/datasets/nogabenyoash/SecQue) dataset (filters to questions mentioning known companies and converts to Gym format):

```bash
cd environments/finance_sec_search

> python prepare.py

Reviewer: Is there a cli command we could use instead? Also this command breaks (wrong import, see my suggestion to `prepare.py`)

```

This writes `data/secque_questions.jsonl`. Use it as the `input_jsonl_fpath` in step 4 below.

**Note that this dataset is not used for training anywhere and is only used for eval/benchmark purposes.**

> ### 2. Start the vLLM server

Reviewer: ```suggestion
### 2. Start the vLLM server (optional)
```

> Launch a vLLM-compatible model server (e.g. Qwen3-30B-A3B) so the policy and judge endpoints are available.

Reviewer: ```suggestion
Launch a vLLM-compatible model server(s) (e.g. Qwen3-30B-A3B) to use them as the policy and judge endpoints. Make sure to populate `env.yaml` to match your endpoint(s) specification. Alternatively you can use an externally hosted OpenAI-compatible endpoint.
```

### 3. Start the Gym servers

With a local vLLM model server:

```bash
config_paths="responses_api_models/vllm_model/configs/vllm_model.yaml,environments/finance_sec_search/config.yaml"
ng_run "+config_paths=[$config_paths]"
```

Or with an OpenAI-compatible API (e.g. OpenAI, Azure, NIM):

```bash
config_paths="responses_api_models/openai_model/configs/openai_model.yaml,environments/finance_sec_search/config.yaml"
ng_run "+config_paths=[$config_paths]"
```

### 4. Collect rollouts

```bash
ng_collect_rollouts \
  +agent_name=finance_agent \
  +input_jsonl_fpath=environments/finance_sec_search/data/example.jsonl \
  +output_jsonl_fpath=results/finance_sec_search_rollouts.jsonl
```

Add `+limit=1` for a quick single-question test:

```bash
ng_collect_rollouts \
  +agent_name=finance_agent \
  +input_jsonl_fpath=environments/finance_sec_search/data/example.jsonl \
  +output_jsonl_fpath=results/finance_sec_search_rollouts.jsonl \
  +limit=1
```

### Run tests

```bash

> ng_test +entrypoint=environments/finance_sec_search

Reviewer: Fails with: `RuntimeError: Missing pyproject.toml or requirements.txt for uv venv setup in server dir: environments/finance_sec_search`

Reviewer: Actually I think there are no tests defined so this should be just removed from the README

```

## Verification

> Uses LLM-as-judge with a financial grading rubric (0/1/2 scale). Only fully correct answers ([[2]]) receive reward 1.0. The judge prompt and rubric are defined in /prompt_templates.

Reviewer: How the user should do it? Is there  a command they should execute, a file they should inspect?
